src/main.cpp

```diff
@@ -107,6 +110,40 @@ LRESULT CALLBACK WndProc(HWND hwnd, UINT msg, WPARAM wparam, LPARAM lparam) {
 			NvimSendResize(context->nvim, rows, cols);
 		}
 	} return 0;
+	case WM_MOVE: {
+		RECT window_rect;
+		DwmGetWindowAttribute(hwnd, DWMWA_EXTENDED_FRAME_BOUNDS, &window_rect, sizeof(RECT)); // Get window position without shadows
+		HMONITOR monitor = MonitorFromPoint({window_rect.left, window_rect.top}, MONITOR_DEFAULTTONEAREST);
+		UINT current_dpi = 0;
+		GetDpiForMonitor(monitor, MDT_EFFECTIVE_DPI, &current_dpi, &current_dpi);
+		if (current_dpi != context->saved_dpi_scaling) {
+			float dpiScale = (float)current_dpi / (float)context->saved_dpi_scaling;
+			GetWindowRect(hwnd, &window_rect); // Window RECT with shadows
+			int new_window_width = (window_rect.right - window_rect.left) * dpiScale + 0.5f;
+			int new_window_height = (window_rect.bottom - window_rect.top) * dpiScale + 0.5f;
+
+			// Make sure window is not larger than the actual monitor
+			MONITORINFO MonitorInfo;
+			MonitorInfo.cbSize = sizeof(MonitorInfo);
+			GetMonitorInfo(monitor, &MonitorInfo);
+			uint32_t MonitorWidth = MonitorInfo.rcWork.right - MonitorInfo.rcWork.left;
+			uint32_t MonitorHeight = MonitorInfo.rcWork.bottom - MonitorInfo.rcWork.top;
+			if (new_window_width > MonitorWidth) new_window_width = MonitorWidth;
+			if (new_window_height > MonitorHeight) new_window_height = MonitorHeight;
+
+			SetWindowPos(hwnd, nullptr, 0, 0, new_window_width, new_window_height, SWP_NOMOVE | SWP_NOOWNERZORDER);
+
+			context->renderer->dpi_scale = current_dpi / 96.0f;
+			RendererUpdateFont(context->renderer, context->renderer->last_requested_font_size);
+			auto [rows, cols] = RendererPixelsToGridSize(context->renderer,
+				context->renderer->pixel_size.width, context->renderer->pixel_size.height);
+			if (rows != context->renderer->grid_rows || cols != context->renderer->grid_cols) {
+				NvimSendResize(context->nvim, rows, cols);
+			}
+
+			context->saved_dpi_scaling = current_dpi;
+		}
+	} return 0;
```

Comment:
Looks good, there is some inconsistencies in the naming though, please stick to `snake_case` for all variables and parameters, `dpiScale`, `MonitorInfo`, `MonitorWidth` and `MonitorHeight` should be changed. Also please use `static_cast<float>()` instead of C-style casts, I don't have a strong preference, but I've used the C++ style casts elsewhere in this project.